service/matching/physical_task_queue_manager.go

```diff
@@ -295,6 +400,11 @@ func (c *physicalTaskQueueManagerImpl) PollTask(
 
 		task.namespace = c.partitionMgr.ns.Name()
 		task.backlogCountHint = c.backlogMgr.BacklogCountHint
+
+		if task.redirectInfo == nil && (!task.isStarted() || !task.started.hasEmptyResponse()) {
```

Comment:
oh, I saw the other comment thread. okay, that kinda makes sense but I'm a little confused.. we have existing metrics for tasks matched, and we didn't need this `hasEmptyResponse` method before. why not?

Reply:
well I wouldn't fully rely on matching metrics to begin with. But I guess what we are guarding against is a rare race condition because, typically, by the time that the forwarded poll expires, the original poll should be already expired.